homeassistant/components/iotmeter/coordinator.py

```diff
@@ -0,0 +1,108 @@
+"""Module for IoTMeter integration in Home Assistant."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from iotmeter_api import IoTMeterAPI, IotMeterAPIError
+
+from homeassistant.helpers.translation import async_get_translations
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+# from .number import ChargingCurrentNumber
+from .sensor import (
+    ConsumptionEnergySensor,
+    EvseSensor,
+    GenerationEnergySensor,
+    TotalPowerSensor,
+)
```

Comment:
The coordinator should not be in charge of creating and removing entities. That is the task of `async_setup_entry` in the platform files. Check Withings for an example, the best example in there are the measurement sensors